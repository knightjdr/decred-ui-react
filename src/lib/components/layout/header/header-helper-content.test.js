import { shallow } from 'enzyme';

import HeaderHelperContent from './header-helper-content';
import TextTransform from '../../../shared/text-transform';

jest.mock('../../../shared/text-transform');
TextTransform.mockReturnValue('text');

const icon = 'icon';
const text = 'text';
const iconOnly = false;
const textOnly = false;
const textColor = '#000000';
const fontSize = 20;
const fontSizeUpper = 15.6;
const children = null;
const textStyle = {};

describe('HeaderHelperContent', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('Renders with icon and text', () => {
    const wrapper = shallow(HeaderHelperContent(
      icon,
      text,
      iconOnly,
      textOnly,
      textColor,
      fontSize,
      fontSizeUpper,
      children,
      textStyle,
    ));
    expect(wrapper).toMatchSnapshot();
    const iconElement = wrapper.find('.decred-ui-header-icon');
    expect(iconElement.length).toBe(1);
    expect(iconElement.text()).toBe('icon');
    const textElement = wrapper.find('.decred-ui-header-text');
    expect(textElement.length).toBe(1);
    expect(textElement.text()).toBe('text');
    expect(TextTransform).toHaveBeenCalledTimes(1);
    expect(TextTransform).toHaveBeenCalledWith(text, fontSize, fontSizeUpper);
  });

  test('Only renders icon', () => {
    const wrapper = shallow(HeaderHelperContent(
      icon,
      text,
      true,
      false,
      textColor,
      fontSize,
      fontSizeUpper,
      children,
      textStyle,
    ));
    expect(wrapper).toMatchSnapshot();
    const iconElement = wrapper.find('.decred-ui-header-icon');
    expect(iconElement.length).toBe(1);
    const textElement = wrapper.find('.decred-ui-header-text');
    expect(textElement.length).toBe(0);
  });

  test('Only renders text', () => {
    const wrapper = shallow(HeaderHelperContent(
      icon,
      text,
      false,
      true,
      textColor,
      fontSize,
      fontSizeUpper,
      children,
      textStyle,
    ));
    expect(wrapper).toMatchSnapshot();
    const iconElement = wrapper.find('.decred-ui-header-icon');
    expect(iconElement.length).toBe(0);
    const textElement = wrapper.find('.decred-ui-header-text');
    expect(textElement.length).toBe(1);
  });

  test('Renders both icon and text if both set to true', () => {
    const wrapper = shallow(HeaderHelperContent(
      icon,
      text,
      true,
      true,
      textColor,
      fontSize,
      fontSizeUpper,
      children,
      textStyle,
    ));
    const iconElement = wrapper.find('.decred-ui-header-icon');
    expect(iconElement.length).toBe(1);
    const textElement = wrapper.find('.decred-ui-header-text');
    expect(textElement.length).toBe(1);
  });

  test('Renders custom text color', () => {
    const wrapper = shallow(HeaderHelperContent(
      icon,
      text,
      false,
      false,
      '#FD714A',
      fontSize,
      fontSizeUpper,
      children,
      textStyle,
    ));
    const textElementStyle = wrapper.find('.decred-ui-header-text').props().style;
    expect(textElementStyle).toHaveProperty('color', '#FD714A');
  });

  test('Renders with children instead of text', () => {
    const wrapper = shallow(HeaderHelperContent(
      icon,
      text,
      iconOnly,
      textOnly,
      textColor,
      fontSize,
      fontSizeUpper,
      'children',
      textStyle,
    ));
    const textElement = wrapper.find('.decred-ui-header-text');
    expect(textElement.length).toBe(1);
    expect(textElement.text()).toBe('children');
    expect(TextTransform).toHaveBeenCalledTimes(0);
  });

  test('Renders with custom style and this style takes precedent', () => {
    const wrapper = shallow(HeaderHelperContent(
      icon,
      text,
      iconOnly,
      textOnly,
      textColor,
      fontSize,
      fontSizeUpper,
      children,
      {
        color: '#ffff00',
        testProp: 'test',
      },
    ));
    const textElement = wrapper.find('.decred-ui-header-text');
    expect(textElement.length).toBe(1);
    expect(textElement.props().style).toHaveProperty('color', '#ffff00');
    expect(textElement.props().style).toHaveProperty('testProp', 'test');
  });
});
