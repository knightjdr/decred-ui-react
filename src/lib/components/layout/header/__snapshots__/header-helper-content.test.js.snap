// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderHelperContent Only renders icon 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    className="decred-ui-header-content"
>
    <span
        className="decred-ui-header-icon"
    >
        icon
    </span>
</div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="decred-ui-header-icon"
>
          icon
</span>,
        null,
      ],
      "className": "decred-ui-header-content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "icon",
          "className": "decred-ui-header-icon",
        },
        "ref": null,
        "rendered": "icon",
        "type": "span",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="decred-ui-header-icon"
>
            icon
</span>,
          null,
        ],
        "className": "decred-ui-header-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "icon",
            "className": "decred-ui-header-icon",
          },
          "ref": null,
          "rendered": "icon",
          "type": "span",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`HeaderHelperContent Only renders text 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    className="decred-ui-header-content"
>
    <div
        className="decred-ui-header-text"
        style={
            Object {
                "color": "#000000",
                "marginLeft": 0,
                "marginTop": -2,
              }
        }
    >
        text
    </div>
</div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <div
          className="decred-ui-header-text"
          style={
                    Object {
                              "color": "#000000",
                              "marginLeft": 0,
                              "marginTop": -2,
                            }
          }
>
          text
</div>,
      ],
      "className": "decred-ui-header-content",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "text",
          "className": "decred-ui-header-text",
          "style": Object {
            "color": "#000000",
            "marginLeft": 0,
            "marginTop": -2,
          },
        },
        "ref": null,
        "rendered": "text",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <div
            className="decred-ui-header-text"
            style={
                        Object {
                                    "color": "#000000",
                                    "marginLeft": 0,
                                    "marginTop": -2,
                                  }
            }
>
            text
</div>,
        ],
        "className": "decred-ui-header-content",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "text",
            "className": "decred-ui-header-text",
            "style": Object {
              "color": "#000000",
              "marginLeft": 0,
              "marginTop": -2,
            },
          },
          "ref": null,
          "rendered": "text",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`HeaderHelperContent Renders with icon and text 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    className="decred-ui-header-content"
>
    <span
        className="decred-ui-header-icon"
    >
        icon
    </span>
    <div
        className="decred-ui-header-text"
        style={
            Object {
                "color": "#000000",
                "marginLeft": 6,
                "marginTop": -2,
              }
        }
    >
        text
    </div>
</div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="decred-ui-header-icon"
>
          icon
</span>,
        <div
          className="decred-ui-header-text"
          style={
                    Object {
                              "color": "#000000",
                              "marginLeft": 6,
                              "marginTop": -2,
                            }
          }
>
          text
</div>,
      ],
      "className": "decred-ui-header-content",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "icon",
          "className": "decred-ui-header-icon",
        },
        "ref": null,
        "rendered": "icon",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "text",
          "className": "decred-ui-header-text",
          "style": Object {
            "color": "#000000",
            "marginLeft": 6,
            "marginTop": -2,
          },
        },
        "ref": null,
        "rendered": "text",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="decred-ui-header-icon"
>
            icon
</span>,
          <div
            className="decred-ui-header-text"
            style={
                        Object {
                                    "color": "#000000",
                                    "marginLeft": 6,
                                    "marginTop": -2,
                                  }
            }
>
            text
</div>,
        ],
        "className": "decred-ui-header-content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "icon",
            "className": "decred-ui-header-icon",
          },
          "ref": null,
          "rendered": "icon",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "text",
            "className": "decred-ui-header-text",
            "style": Object {
              "color": "#000000",
              "marginLeft": 6,
              "marginTop": -2,
            },
          },
          "ref": null,
          "rendered": "text",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
